platform/mellanox/mlnx-platform-api/sonic_platform/sfp.py

```diff
@@ -443,15 +448,17 @@ def get_presence(self):
         presence_sysfs = f'/sys/module/sx_core/asic0/module{self.sdk_index}/hw_present' if self.is_sw_control() else f'/sys/module/sx_core/asic0/module{self.sdk_index}/present'
         if utils.read_int_from_file(presence_sysfs) != 1:
             return False
-        eeprom_raw = self._read_eeprom(0, 1, log_on_error=False)
+        eeprom_raw = self.read_eeprom(0, 1, log_on_error=True)
```

Comment:
@tshalvi why do we need to mix hardware presence (which is what `get_presence()` is meant to return) with i2c read ? What is the motivation behind?

Reply:
The motivation is that get_presence() returning True (presence = 1) only indicates that the hardware module is physically detected. It doesn’t necessarily mean that the module’s EEPROM is ready for access. The expectation is to report present=True only when both the hardware is connected and the EEPROM is ready and accessible.